### PR TITLE
Arnaus weather app.

### PR DESCRIPTION
**Pull request by vidalhuix**
## Netlify link
Netlify link [here](https://this-is-my-weather-app.netlify.app/).

## Collaborators
Individual project this time.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  const descrip = data.weather[0].description
+  selectWeatherDescription.innerText = descrip.charAt(0).toUpperCase() + descrip.slice(1)
+
+  
+  //set sunrise "the variable cascade hell-1"
+  const selectSunraise = document.getElementById("sunrise")
+  const sunriseTimeStamp = data.sys.sunrise*1000
+  const sunriseTime = new Date (sunriseTimeStamp)
+  const sunriseHours = sunriseTime.getHours()
+  const sunriseMinutes = sunriseTime.getMinutes()
```
Comment (ericamechler) [inline script.js:53]: maybe this could be explained a bit more for the reader?

File: script.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      <label>${dayOfWeekString}</label>
+      <div class="forecast-day-right">
+      <img src="https://openweathermap.org/img/wn/${weatherIcon}.png"></img>
+      <i>${roundMaxTemp} / ${roundMinTemp} °C</i>
+      </div>
+      </div>`;
+      
+      // Append the HTML content to the forecast element
+      selectForecast.innerHTML += forecastHTML;
+    });
```
Comment (ericamechler) [inline script.js:125]: for consistency, this code could be put into the const updateHTML 2

File: script.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+//https://api.openweathermap.org/data/2.5/weather?q=Stockholm,Sweden&units=metric&APPID=69cc94dd08e05801b495ee1b7a9cba03
+const URL1 = `${BASE_URL}weather?q=${cityCountry}&units=metric&APPID=${API_KEY}`
+
+//https://api.openweathermap.org/data/2.5/forecast?q=Stockholm,Sweden&units=metric&APPID=69cc94dd08e05801b495ee1b7a9cba03
+const URL2 =  `${BASE_URL}forecast?q=${cityCountry}&units=metric&APPID=${API_KEY}`
+
+
+//HERE TODAYS WEATHER INFORMATION FUNCTION ---------------------------------------------------------------------------------------------------------------------//
+const updateHTML1 = (data) =>{
```
Comment (ericamechler) [inline script.js:25]: maybe this could have a clearer name? Like const updateTodaysWeather or so?

File: script.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    selectBackgroundCircle.style.transform = "translate(-50%, -99%)"
+    selectButton.style.top = "43%"
+  } else {
+    selectArrow.style.borderWidth = "6px 0 0 6px"
+    selectBackgroundCircle.style.transform = "translate(-50%, -50%)"
+    selectButton.style.top = "75%"
+  }
+}
+
+selectButton.addEventListener("click", dropdownForecast)
```
Comment (ericamechler) [inline script.js:158]: comment for this one? :)

File: script.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+//DOM selectors-------------------------------------------------------------------------------------------------------------------------------------------------//
+const selectButton = document.getElementById("containerBtn")
+const selectArrow = document.getElementById("arrow")
+const selectBackgroundCircle = document.getElementById("backgroundCircle")
+
+//HERE BUTTON AND FALLING DOWN HEADER BACKGROUND ------------------------------------------------------------------------------------------------------------//
+let isOpen = false //initial state
+
+const dropdownForecast =()=>{
```
Comment (ericamechler) [inline script.js:144]: maybe more explanations for this? :)

File: style.css [hunk 6]
```diff
@@ -0,0 +1,246 @@
+/* google fonts */
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@500&family=Open+Sans:ital,wght@0,300..800;1,300..800&family=Poppins:wght@100;400&family=Roboto:wght@300&display=swap');
+
+:root{
```
Comment (ericamechler) [inline style.css:4]: very nice with the root-system!

File: style.css [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  margin: 5px 0;
+}
+
+.temperature{
+  display: flex;
+  flex-direction: row;
+  justify-content: left;
+}
+
+.t{
```
Comment (ericamechler) [inline style.css:62]: this class name is a bit unclear. Could it become clearer?

File: style.css [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+.forecast-day i{
+  width: 120px;
+  text-align: end;
+}
+
+/* -------------------------------------FOOTER----------------------------------------------- */
+
+footer{
```
Comment (ericamechler) [inline style.css:186]: the footer is very small. Maybe make it a bit higher?

File: index.html [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="./style.css">
+  <title>My weather app</title>
+</head>
+<body>
+  <header>
+    <nav>
+<!--       <input type="text" placeholder="City">
+ -->    </nav>
+ <img class="todays-icon" id="todays-icon"></img>
```
Comment (ericamechler) [inline index.html:14]: this icon becomes blurry in all screen sizes, maybe it's better to have another one that is made to be bigger?

File: style.css [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+footer a{
+  text-decoration: none;
+  font-size: 0.675em;
+  color: white;
+  padding: 10px;
+}
+
+/* -------------------------------------MEDIA QUERIES----------------------------------------------- */
```
Comment (ericamechler) [inline style.css:217]: well done with the media queries!

**Review by ericamechler — COMMENTED** [review]
Very well done Arnau! Love that you got the container-button working :) 

Most of the code is very easy to understand, but some structure-changes could be made as described in the comments. Also, some more comments could be added to describe more to the reader what the code does :) 
You followed the requirements very well!

**Review by AntonellaMorittu — APPROVED** [review]
Hi Arnau, very good job this week! Your code is well structured, formatted, clean and tidy. The app is working well, you completed the basic requirements to pass this assignment. One comment is about the forecast:

- The app should have: 4-day forecast

You are displaying all 5 days when we requested 4, not a big deal. Fix this and you're good to go:) This will also fix the scrolling that in this type of design should be disabled (open the panel but don't allow inner scrolling). 

Keep up the great work 🎸

**Pull request closed without merging**